lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,650 @@
+                sequence_number = sequence_number,
+                total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        if sequence_number >= total_chunks {
+            warn!(
+                message = "Received a chunk with a sequence number greater than total chunks. Ignoring it.",
+                message_id = message_id,
+                sequence_number = sequence_number,
+                total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        let mut state_lock = self.state.lock().unwrap();
+
+        if state_lock.len() >= self.pending_messages_limit {
+            warn!(
+                message = "Received a chunk but reached the pending messages limit. Ignoring it.",
+                message_id = message_id,
+                sequence_number = sequence_number,
+                pending_messages_limit = self.pending_messages_limit,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        let message_state = state_lock.entry(message_id).or_insert_with(|| {
+            // We need to spawn a task that will clear the message state after a certain time
+            // otherwise we will have a memory leak due to messages that never complete
+            let state = Arc::clone(&self.state);
+            let timeout = self.timeout.clone();
+            let timeout_handle = tokio::spawn(async move {
+                tokio::time::sleep(timeout).await;
+                let mut state_lock = state.lock().unwrap();
+                if let Some(_) = state_lock.remove(&message_id) {
+                    // TODO: record metrics here? it would be insteresting to know how many messages are being discarded
+                    let message = format!("Message was not fully received within the timeout window of {}ms. Discarding it.",
+                        timeout.as_millis());
+                    warn!(
+                        message = message,
+                        message_id = message_id,
+                        timeout = timeout.as_millis(),
+                        internal_log_rate_limit = true
+                    );
+                }
+            });
+            MessageState::new(total_chunks, timeout_handle)
+        });
+
+        if message_state.total_chunks != total_chunks {
+            warn!(message_id = "Received a chunk with a different total chunks than the original. Ignoring it.",
+                message_id = message_id,
+                original_total_chunks = message_state.total_chunks,
+                received_total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        if message_state.is_chunk_present(sequence_number) {
+            // TODO: add a PR comment asking for info or warn
+            info!(
```

Comment:
`info` level because a duplicate packet is not really a decoding error, it can be safely ignored (assuming contents were equal, but we are not checking it)

Reply:
As we are dropping the data, maybe this should also return an `Err`?